Implemented fixes as per the Task SCRUM 8

saurabh-pingale-cognito opened the pull request.

saurabh-pingale committed:
Implemented fixes as per the Task SCRUM 8

BalanaguYashwanth commented on frontend/warehouse-app/src/contexts/AddressContext.tsx:
```diff
@@ -67,6 +71,19 @@ const initialState: AddressState = {
   error: null,
 };
 
+const mapCourierToAddress = (courier: any): AddressData => {
```
I am seeing ``??`` and ``""`` rather than code

BalanaguYashwanth commented on frontend/warehouse-app/src/contexts/AddressContext.tsx:
```diff
@@ -171,10 +209,35 @@ export const AddressProvider: React.FC<AddressProviderProps> = ({ children }) =>
   };
 
+  const loadUserPreferences = async (userId: string) => {
+    dispatch({ type: "LOAD_USER_PREFERENCES_START" });
```
what do you mean by ``START``
LOAD_USER_PREFERENCES_START

BalanaguYashwanth committed:
resolved creating package bug of create query using user_preferences

BalanaguYashwanth committed:
adjust cosmetic changes and add package.json dev command for quick start

saurabh-pingale committed:
Updated changes from the Task Scrum 8

saurabh-pingale committed:
New changes mentioned in the Task SCRUM 8

saurabh-pingale committed:
Fixed the flow

BalanaguYashwanth committed:
implemented ui level of dynamic buttons according to status

saurabh-pingale committed:
Refactor package UI and logic:

- Updated button naming and added dynamic buttons for all package statuses
- Added warning banner when no items are added in a package
- Prevent status change to "Ready To Send" if no items are added
- Changed return status format to { status: { label: "", value: "" } }
- Fixed UI for uploading PDFs: added thumbnail and restricted file name to 10 characters
- Corrected status checkbox behavior for statuses beyond "Ready To Send"
- Made volumetric weights optional in register modal
- Fixed auto logout on page refresh
- Minor code improvements

saurabh-pingale committed:
Refactored code and fixed UI issues

BalanaguYashwanth committed:
Merge pull request #29 from Cognito-Innovations/fix/refactor-code

Refactored code and fixed UI issues

BalanaguYashwanth committed:
Merge pull request #28 from Cognito-Innovations/feature/uiimprove-actionbuttons-status

Feature/uiimprove actionbuttons status

saurabh-pingale committed:
Fix UI issues in Shopping and Pickup Request, refactor tables into a common component, add Tracking Request in Pickup Request, and fix minor bugs

BalanaguYashwanth committed:
add todos in backend and frontend - trackingrequests and others

saurabh-pingale committed:
Split the components to make it modularize and resolved minor issues.

BalanaguYashwanth committed:
Merge pull request #31 from Cognito-Innovations/refactor/split-components

Split the components to make it modularize and resolved minor issues.

BalanaguYashwanth committed:
Merge pull request #30 from Cognito-Innovations/refactor-code

Fix UI issues in Shopping and Pickup Request, refactor tables into a common component, add Tracking Request in Pickup Request, and fix minor bugs

BalanaguYashwanth committed:
resolve linting issues and api response followed with dto and other api service efficiency code changes

BalanaguYashwanth committed:
removed unuse components and sections

BalanaguYashwanth committed:
corrected api and removed unuse

saurabh-pingale committed:
Added pdf generation for packages, refactored the ui.

saurabh-pingale committed:
Added dockerfile and app.yaml

saurabh-pingale committed:
Fix the countries code

BalanaguYashwanth committed:
replace brandname and add address and removed password section

BalanaguYashwanth committed:
fix price flows, pdf and other bugs in shopping request

BalanaguYashwanth committed:
removed notification, improved printlabelpdf

BalanaguYashwanth committed:
add flow of package charges of shipping inovices

BalanaguYashwanth committed:
improved the shipping export requsts and remove depart in action status buttons

BalanaguYashwanth committed:
update logos in login and homescreen

BalanaguYashwanth committed:
resolve lint issues and lib lucide issues and build to deploy

BalanaguYashwanth committed:
fix the country update in admin and print carrier label pdf

saurabh-pingale committed:
testing code

saurabh-pingale committed:
Moved password hashing to backend, fixed visibility icons

BalanaguYashwanth committed:
configured backend url

BalanaguYashwanth committed:
resolve login google auth issue

saurabh-pingale committed:
Fixed minor issues

BalanaguYashwanth committed:
Merge pull request #33 from Cognito-Innovations/vercel-deployment

Moved password hashing to the backend, fixed register and login with credentials and fixed visibility icons for password.

BalanaguYashwanth committed:
Merge pull request #32 from Cognito-Innovations/implementation/pdf-refactor-ui

Added pdf generation for packages, refactored the ui.

BalanaguYashwanth committed:
Merge pull request #34 from Cognito-Innovations/improvements/api-frontend-component

Improvements/api frontend component

saurabh-pingale committed:
- Added loader on Receive button and disabled it if status is "receive" in pre-arrival
- Replaced search by item name with shopping request ID in client
- Fixed reject button visibility in shopping request admin
- Optimized shopping request detail load time to ~5s
- Fixed shopping request not showing on refresh in client
- Added proper gaps between loaders and text in buttons
- Fixed auto-populate form on edit in shipment export

BalanaguYashwanth committed:
Replaced promise.all to promise.allsettled and improve flow of upload in package

BalanaguYashwanth committed:
update the remarks in request to ship in package section

BalanaguYashwanth committed:
update vite config

BalanaguYashwanth committed:
resolve bug of email template .hbs

saurabh-pingale committed:
Add Playwright tests for admin flows, improve client UI and fix shopping request card

saurabh-pingale committed:
Fix: resolved TS errors, improved validation, and UI updates

BalanaguYashwanth committed:
Merge pull request #38 from Cognito-Innovations/fix/ts-errors-validation-ui-updates

Fix: resolved TS errors, improved validation, and UI updates

saurabh-pingale committed:
Fix TypeScript errors and register package

saurabh-pingale committed:
feat: add history tab, package display, prearrival popup with delete, search bar, and suite number display

BalanaguYashwanth committed:
Merge pull request #40 from Cognito-Innovations/feature/mysuite-history-prearrival-search

feat: add history tab, package display, prearrival popup with delete, search bar, and suite number display

BalanaguYashwanth committed:
Merge pull request #39 from Cognito-Innovations/fix/ts-errors-register-package

Fix TypeScript errors and register package

BalanaguYashwanth committed:
fix bug of otp history tab count display

saurabh-pingale committed:
Refactor PreArrival service, fix Search bar, and Shopping Request issues

BalanaguYashwanth committed:
resolve bug in display of package detail of user details

BalanaguYashwanth committed:
resolve bug in pickup requests

BalanaguYashwanth committed:
add validation to the DTOs

BalanaguYashwanth committed:
Merge pull request #41 from Cognito-Innovations/feature/refactor-prearrival-search-quotation

Refactor PreArrival service, fix Search bar, and Shopping Request issues

BalanaguYashwanth committed:
Merge pull request #37 from Cognito-Innovations/tests/playwright

Add Playwright tests for admin flows, improve client UI and fix shopping request card

BalanaguYashwanth committed:
Merge pull request #36 from Cognito-Innovations/improvements/top-of-shopping-reques-improvements

Replaced promise.all to promise.allsettled and improve flow of upload…

BalanaguYashwanth committed:
Merge pull request #35 from Cognito-Innovations/feature/shopping-request-improvements

feat: improve shopping request workflows and UI/UX

saurabh-pingale committed:
Enhance package handling, improve UI previews, and fix key issues

BalanaguYashwanth committed:
resolved issue in carieer label pdf

saurabh-pingale committed:
Added the loader while verifying the photos/docs, fixed UI for pickup equest for pickup_address. Limit the inputs while creating the pickup request.

saurabh-pingale committed:
Fixed backend error display on the frontend for package registration

saurabh-pingale committed:
Updated tracking number validation range to 5–15 digits

saurabh-pingale committed:
- Reverted the functionality that changed the status from "Action Required" to "Ready To Send" when a Photo/Document was uploaded on the Package Admin side.
- Fixed the loading state handling of the "Request Ship" button in the Packages tab on the Client side.

BalanaguYashwanth committed:
Merge pull request #43 from Cognito-Innovations/fix/display-error-register-package

Fixed backend error display on the frontend for package registration

saurabh-pingale committed:
Merge branch 'feature/package-enhancements-and-bug-fixes' of https://github.com/Cognito-Innovations/warehouse into fix/action-logs-loder

saurabh-pingale committed:
fix: improve upload handling and validation for non-image files

BalanaguYashwanth committed:
Merge pull request #44 from Cognito-Innovations/fix/action-logs-loder

Added loader for photo/doc verification and improved Pickup Request UI

BalanaguYashwanth committed:
Merge pull request #42 from Cognito-Innovations/feature/package-enhancements-and-bug-fixes

Enhance package handling, improve UI previews, and fix key issues

BalanaguYashwanth committed:
add err catch block in the actionlogs section

BalanaguYashwanth committed:
add ecommerce module and rename products to shopping request products

BalanaguYashwanth committed:
update naming convertion of ecommerce module all files - categories, sub_categories and products

BalanaguYashwanth committed:
implement basic ecommerce flow and cart apis

BalanaguYashwanth committed:
add todos

BalanaguYashwanth committed:
add next set of todos in backend

saurabh-pingale committed:
Added ecommerce menu and measurement integration

saurabh-pingale committed:
Removed unnecessary code,
improved code, replaced discount_amount with discount_percentage in admin/backend.

saurabh-pingale committed:
Replaced discount_amount with discount_percentage and mock data with actual API in client, made category and product controllers public, added remaining fields in forms for Category, SubCategory, Product.

saurabh-pingale committed:
Removed mapper from PackageDetail and updated associated code to use data from API Response, added constant values in constants, splitted code in components

saurabh-pingale committed:
Removed mappers from the backend, added controller, service, entity and dto for shipment, removed shipment related code from the packages, added shipment UI in client and admin side.

saurabh-pingale committed:
Updated the UI, controller, service, entity and dto for the shipments.

saurabh-pingale committed:
Progressed Shipment functionality:
- Added remaining UI for Shipment Admin
- Updated code to use Shipment instead of package in Shipment Export
- Removed some shipment related code from packages
- Updated code to follow the shipment API response

saurabh-pingale committed:
Fixed mentioned changes

saurabh-pingale committed:
- fixed small UI chages in shipment detail
- fixed ecommerce (add to cart, place order)
- added order in the client

saurabh-pingale committed:
Fixed Code:
-  Discount calculation
- Items getting added to cart while refresh
- Restrict add to cart till stock quantity
- If no available product then show product as "Out of Stock"
- Improve checkout page UI
- Resolve bugs from cliq history messages

saurabh-pingale committed:
Fixed bugs:
- fixed API
- fixed status color
- fixed circular structure error

saurabh-pingale committed:
Added elastic beanstalk and sh files to work with the AWS beanstalk

rohit133 committed:
Shopfront refactor -> Cart, Product Page and Home page

BalanaguYashwanth committed:
add aws elasticbeam workflow for nestjs

BalanaguYashwanth committed:
readme change to trigger github workflow

BalanaguYashwanth committed:
recorrct in github workflow yaml

BalanaguYashwanth committed:
recorrect in github workflow yaml

BalanaguYashwanth committed:
recorrect package manager pnpm in github workflow yaml

BalanaguYashwanth committed:
add pnpm.yaml file for github workflow

BalanaguYashwanth committed:
correct the port and add health api

BalanaguYashwanth committed:
add procfile for correct install in github actions

BalanaguYashwanth committed:
add nvmrc for correct deployment refernce version

BalanaguYashwanth committed:
add script with procfile for aws elastic beamstack

BalanaguYashwanth committed:
add category to the admin ecommerce section

BalanaguYashwanth committed:
revert platform files and add new github action workflows

BalanaguYashwanth committed:
moved github workflow inside the warehouse server

BalanaguYashwanth committed:
moved github workflow to root

BalanaguYashwanth committed:
add pnpm installation in workflow yml

BalanaguYashwanth committed:
correct the app name and env name in workflow.yml

BalanaguYashwanth committed:
revert package start command in nestjs warehouse server

BalanaguYashwanth committed:
add cloudinary in package backend server

BalanaguYashwanth committed:
add correct start command in server package.json file

BalanaguYashwanth committed:
add cloudinary in package json server file

BalanaguYashwanth committed:
update procfile

BalanaguYashwanth committed:
update ecommerce front main page header

BalanaguYashwanth committed:
fix workflow yml file

BalanaguYashwanth committed:
remove unuse file of zip, add only backend server files modified in workflow yml file

BalanaguYashwanth committed:
removed nooptional in action worklow yml file

BalanaguYashwanth committed:
final:update workflow yml to handle css or assets files

BalanaguYashwanth committed:
add tslib lib to package.json

BalanaguYashwanth committed:
add tslib lib to package-lock.json

BalanaguYashwanth committed:
add uid lib to server for worflow.yml to work

BalanaguYashwanth committed:
add express, load-esm and multer to server

BalanaguYashwanth committed:
add iterare to server

BalanaguYashwanth committed:
trigger to push small readme change

BalanaguYashwanth committed:
Merge pull request #50 from Cognito-Innovations/feature/shopfront-refactor

Shopfront refactor -> Cart, Product Page and Home page

BalanaguYashwanth committed:
improved skeleton loader for ecommerce page and cargo_type to seeddata and installed fast string lib for aws beamstack

saurabh-pingale committed:
Added missing libraries and added nod_modules in the .ebignore

saurabh-pingale committed:
Added loadsh dependency

saurabh-pingale committed:
Added js-yaml dependency

saurabh-pingale committed:
Added @nestjs/mapped, @nestjs/swagger, class-transfoermer and class-validator dependencies

saurabh-pingale committed:
Added ansis dependency

saurabh-pingale committed:
Added sql-highlight dependency

saurabh-pingale committed:
Added @sqltools/formatter dependency

saurabh-pingale committed:
Added app-root-path dependency

saurabh-pingale committed:
Added sha.js dependency

saurabh-pingale committed:
Added inherits dependency

saurabh-pingale committed:
Added safe-buffer dependency

saurabh-pingale committed:
Added to-buffer dependency

saurabh-pingale committed:
Added isarray dependency

saurabh-pingale committed:
Added typed-array-buffer dependency

saurabh-pingale committed:
Added es-errors/type dependency and updated .npmrc to force install all dependencies

saurabh-pingale committed:
Added call-bound dependency

saurabh-pingale committed:
Added EB platform hooks for pnpm

saurabh-pingale committed:
Added shamefully-hoist to .npmrc and updated the hook script

saurabh-pingale committed:
Fix EB hooks: correct Bash content, add cd/logging, enforce hoisting for pnpm

saurabh-pingale committed:
Added hoisted layout in .npmrc

saurabh-pingale committed:
Added get-intrinsic dependency

saurabh-pingale committed:
Removed hosting layout from .npmrc

saurabh-pingale committed:
Fixed pnpm-lock.yaml

saurabh-pingale committed:
Added get-intrinsic dependency

saurabh-pingale committed:
- Added multi select cuntries
- Fixed authentication issues on home and product details page
- Added "Login" button if user not logged; added "Add Address" button if user loggd In and no address; else show address
- Mapped categories in promtional card
- Aligned product details tab
- Fixed offer cards to show in collapse if more than one card is there else show simply card

BalanaguYashwanth committed:
update latest workfile.yml

BalanaguYashwanth committed:
Merge branch 'fix/bugs' of https://github.com/Cognito-Innovations/warehouse into fix/bugs

BalanaguYashwanth committed:
removed ignore and platform hooks file

BalanaguYashwanth committed:
correct zip path structure in workflow.yml

BalanaguYashwanth committed:
add correct zip path in workflow.yml

BalanaguYashwanth committed:
add correct pnpm install package run command in workflow.yml

BalanaguYashwanth committed:
removed prod in workflow.yml

BalanaguYashwanth committed:
revert yml file

BalanaguYashwanth committed:
update wokflow yml file

BalanaguYashwanth committed:
correct the api base url of ecommerce api

saurabh-pingale committed:
Refactord Code:
- Added cancel for removing the selected countries
- Added loder while fetching the countries in categories, sub-categories and products
- Added wrapper to return the price based on country and used in ecommerce-product service
- Added cargo-options and set up relationship with the ecommerce-categories
- Added mobile responsiveness for the ecommerce (home, products details, cart)
- Improved code

BalanaguYashwanth committed:
Merge pull request #51 from Cognito-Innovations/refactor/design-backend-admin-changes

Refactor design improvements

saurabh-pingale committed:
Refactor Code:
- Fixed scroll button for desktop screen
- Fixed image height/width in product details page
- Fixed price fluctuating in product details page
- Added "All" promotional card
- Added search functionality
- Fixed synching price between cart and checkout
- Added edit, delete and toggle for product in admin
- Replace Image URL with the Image uploader
- Improve code

BalanaguYashwanth committed:
add todods in ecommerce products

saurabh-pingale committed:
Refactor Code:
- Fixed location detection (detected by IP, fallback as geo location)
- Added infinite scrolling using Intersection Oberserver API
- Fixed API calls on Ecommerce page when redirect back
- Preserve product and categories in zustand store until page refresh
- Added "View Order History" button in Order Success Popup and redirect to order page on dashbaord
- Removed Ids in category, subcategory and products in admin
- Added rate limit using throttle
- Added IP detection middleware, entity, service and interceptor to get the IP from request, store it in DB if not exist and return in headers
- Added interceptor to attach id to each request to track IP

saurabh-pingale committed:
Added payment flow

saurabh-pingale committed:
Fixed bugs:
- Fixed skeleton loaders to wrok partially
- Fixed skeleton loader in the cart page
- Fixed add to cart, increment, decrement and delete in cart page
- Fixed cashfree payment gateway in backend and frontend

saurabh-pingale committed:
- Fixed total amount to convert by order currency before passing to cashfree
- Removed OrderSuccess Popup to display after paymant successful because we're redirecting to the order in dashbaord
- Fixed add address functionality if there is no address available

BalanaguYashwanth committed:
Merge pull request #54 from Cognito-Innovations/fix/bugs-cashfree

Fixed bugs:

BalanaguYashwanth committed:
Merge pull request #53 from Cognito-Innovations/config/cashfree

Added payment flow

BalanaguYashwanth committed:
Merge pull request #52 from Cognito-Innovations/refactor/warehouse-design-code

Refactor Code:

BalanaguYashwanth committed:
Merge pull request #49 from Cognito-Innovations/fix/bugs

Fixed bugs:

BalanaguYashwanth committed:
Merge pull request #48 from Cognito-Innovations/feature/ecommerce-menu

Added ecommerce menu and measurement integration

BalanaguYashwanth committed:
Merge pull request #47 from Cognito-Innovations/improvements/prority-todos

Add priority todos

BalanaguYashwanth committed:
Merge pull request #46 from Cognito-Innovations/todos/ecommerce-frontend-cart-apis

implement basic ecommerce flow and cart apis
